properties/handlers.py

```diff
@@ -2,34 +2,37 @@
 from six import string_types
 
 
-def _set_listener(instance, observer):
-    """
-        Add listeners to a Properties class instance
-    """
+def _set_listener(instance, obs):
+    """Add listeners to a Properties class instance"""
 
-    for name in observer.names:
+    for name in obs.names:
         if name not in instance._listeners:
-            # TODO: extend to different kinds of observers
-            # instance._listeners[name] = {observer.kind: [observer]}
-            instance._listeners[name] = [observer]
-        else:
-            instance._listeners[name] += [observer]
+            instance._listeners[name] = {'validate': [], 'observe': []}
+        instance._listeners[name][obs.mode] += [obs]
```

Comment:
`observe.always` is called on set even when the given value isn't different, whereas `observe.change` is called only when the value is different? Both are called after the new value is set though?

Then, using the `@validator` is only used for methods to be called on `validate()`?